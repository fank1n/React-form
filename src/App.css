html{
  margin: 0;
}
body{
  background-color: rgb(255, 255, 255);
}
.container{
  margin: 0 auto;
  width: 1200px;
  height: 1000px;
  background-color: #fff;
  box-shadow: 0px 5px 30px 1px rgba(34, 60, 80, 0.6);
}
header{
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  background-color: rgb(112, 173, 201);
  border-bottom-left-radius: 10px;
  border-bottom-right-radius: 10px;
  box-shadow: 0px 5px 10px 2px rgba(34, 60, 80, 0.2);
}
#header-name{
  text-align: center;
  font-size: 26px;
  font-weight: bold;
  margin: 0;
}
.page-ref{
  margin: 30px;
  text-decoration: none;
  color: rgb(0, 0, 0);
}
.lil-container{
  display: flex;
}
.item-container{
  margin: 0 20px;
  border-bottom: 2px solid black;
}
.item-container-disabled{
  margin: 0 20px;
}
.login-btn{
  align-self: center;
  width: 60px;
  height: 30px;
  background-color: black;
  color: #fff;
  border: none;
  cursor: pointer;
  border-radius: 5px;
}
.login-btn:hover{
  border: 2px solid white
}
.inner-cont{
  display: flex;
  gap: 800px;
  padding-top: 20px;
}
.cart-info{
  margin: 0;
}
.prod-price{
  font-weight: bold;
}
