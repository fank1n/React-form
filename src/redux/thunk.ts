// import {getTodos} from "../api/todos";
import INITIALIZE_TODOS from "./actions/todoActions"

export interface ListItem {
    "id": number,
    "description": string,
    "completed": false,
    "favorite": false
}

export const getTodos = () => {
    return async (dispatch:any) => {
        const todos = await fetch ("http://localhost:8000/todo")
        .then((todos) => todos.json)
        .then((todos) => {
            dispatch({
                type: INITIALIZE_TODOS,
                payload: todos,
            })
        }).catch((error) => console.log(error))
    }
}

