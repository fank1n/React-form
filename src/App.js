import "./App.css";
import {
  BrowserRouter as Router,
  Switch,
  Route,
  NavLink,
} from "react-router-dom";
import Home from "./homePage/Home";
import About from "./aboutPage/About";
import Description from "./descriptionPage/Descriptrion";
import Modal from "./modal/Modal";
import { useState, useEffect } from "react";
import Context from "./context";

const App = () => {
  const [modalActive, setModalActive] = useState(false);
  const [cartInfo, setCartInfo] = useState([]);
  const [inStock, setInStock] = useState([]);
  const [isAddedProduct, setIsAddedProduct] = useState(true);
  const [succesLogin, setSuccesLogin] = useState(false);
  const [formValid, setFormValid] = useState(false);
  const [isAdmin, setIsAdmin] = useState(false);
  const [loggedInfo, setLoggedInfo] = useState({
    userData: [
      {
        login: "admin",
        password: "admin",
      },
      {
        login: "customer",
        password: "customer",
      },
    ],
  });

  useEffect(() => {
    fetch("http://localhost:8000/products")
      .then((res) => res.json())
      .then((data) => setInStock(data));
  }, []);

  useEffect(() => {
    fetch("http://localhost:8000/cart")
      .then((res) => res.json())
      .then((data) => setCartInfo(...data));
  }, []);

  const handlerPutCart = async (product, prodCount = 1) => {
    setIsAddedProduct(() => !isAddedProduct);
    let [inStockItem] = [inStock.find((item) => item.id === product.id)];
    inStockItem.inStock -= prodCount;
    setInStock((previous) => {
      return [
        ...previous.filter((item) => item.id !== product.id),
        inStockItem,
      ];
    });
    setCartInfo((previous) => {
      return {
        ...previous,
        count: Number(previous.count) + Number(prodCount),
        amount: Number(previous.amount) + Number(product.price),
      };
    });

    fetch(`http://localhost:8000/products/${product.id}`, {
      method: "PUT",
      headers: { "Content-Type": "application/json" },
      body: JSON.stringify({
        id: product.id,
        picture: product.picture,
        title: product.title,
        price: product.price,
        inStock: inStockItem.inStock,
        fullDescription: product.fullDescription,
      }),
    })
      .then((res) => {
        return res.json();
      })
      .catch((error) => {
        console.log(error);
      });
  };

  fetch(`http://localhost:8000/cart/1`, {
    method: "PUT",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify({
      count: 0,
      amount: 0,
    }),
  })
    .then((res) => {
      return res.json();
    })
    .catch((error) => {
      console.log(error);
    });

  return (
    <Context.Provider
      value={{
        cartInfo,
        setCartInfo,
        handlerPutCart,
        inStock,
        isAddedProduct,
        loggedInfo,
        setLoggedInfo,
        succesLogin,
        setSuccesLogin,
        formValid,
        setFormValid,
        isAdmin,
        setIsAdmin
      }}
    >
      <Router>
        <div className="App">
          <div className="container">
            <header>
              <div className="inner-cont">
                {!formValid ? (
                  <button
                    className="login-btn"
                    onClick={() => {
                      setModalActive(true);
                    }}
                  >
                    login
                  </button>
                ) : (
                  <button
                    className="login-btn"
                    onClick={() => {
                      setModalActive(true);
                      setFormValid(false);
                    }}
                  >
                    logout
                  </button>
                )}
                {formValid ? (
                  <div className="outer-cart-info-container">
                    <div className="cart-container">
                      <p className="cart-count">
                        Товаров в корзине: {cartInfo.count} шт.
                      </p>
                      <p className="cart-amount">
                        На сумму: {cartInfo.amount} руб.
                      </p>
                    </div>
                  </div>
                ) : (
                  <div className="outer-cart-info-container"></div>
                )}
              </div>
              <div className="lil-container">
                <div className="item-container">
                  <NavLink
                    exact
                    to="/"
                    className="page-ref"
                    activeClassName="page-ref-active"
                  >
                    Home
                  </NavLink>
                </div>
                <div className="item-container">
                  <NavLink
                    exact
                    to="/about"
                    className="page-ref"
                    activeClassName="page-ref-active"
                  >
                    About
                  </NavLink>
                </div>
              </div>
            </header>
            <Modal active={modalActive} setActive={setModalActive}></Modal>
            <Switch>
              <Route exact path="/" component={Home} />
              <Route exact path="/about" component={About} />
              <Route exact path="/product/:productID" component={Description} />
              <Route render = {() => <h1 className = "not-found"> Упс, страница не найдена</h1>}/>
            </Switch>
          </div>
        </div>
      </Router>
    </Context.Provider>
  );
};

export default App;
