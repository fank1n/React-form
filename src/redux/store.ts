import { applyMiddleware, createStore, compose } from 'redux';
// import { configureStore } from "@reduxjs/toolkit";
import todosReducer from "./reducers/addTodoReducer";
import thunk from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension';
// export const store = configureStore({
//   reducer: todosReducer,
// });

// export const store = createStore(
//   todosReducer, 
//   compose(applyMiddleware(ReduxThunk),     
//   window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__())
// );
// export const store = createStore(
//   todosReducer,
//   compose(
//       applyMiddleware(thunk),
//   )
// )
export const store = createStore(todosReducer, composeWithDevTools(applyMiddleware(thunk)));

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;