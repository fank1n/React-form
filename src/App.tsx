import React from 'react';
import { useSelector } from 'react-redux';
import './App.css';
import { TodoInput } from './containers/components/TodoInput/TodoInput';
import { TodoItemsList } from './containers/components/TodoList/TodoItemsList';
import { RootState } from './redux/store';

function App() {
  const todoList = useSelector((state: RootState) => state);
  return (
    <div className="App">
      <TodoInput/>
      <TodoItemsList/>
    </div>
  );
}

export default App;
