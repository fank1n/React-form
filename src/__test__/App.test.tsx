import App from '../App';
import {describe, expect} from '@jest/globals'
import {mount} from "enzyme"
import { Provider } from 'react-redux';
import { store } from '../redux/store';

describe('App', () => {
  
  let wrapper:any;

  beforeEach(() => {
    wrapper = mount(<Provider store={store}><App/></Provider>)
  });
  
  afterEach(() => {
    wrapper.unmount()
  });

  it('App component snapshot should be correct', () => {
    expect(wrapper).toMatchSnapshot()
  });

  it('TodoInput component should render correctly',() => {
    expect(wrapper.find('TodoInput')).toBeTruthy()
  });

  it('TodoItemsList component should render correctly',() => {
    expect(wrapper.find('TodoItemsList')).toBeTruthy()
  });
});
