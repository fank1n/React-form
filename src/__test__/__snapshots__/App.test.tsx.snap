// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App App component should render correctly 1`] = `ReactWrapper {}`;

exports[`App App component snapshot should be correct 1`] = `ReactWrapper {}`;

exports[`App component should render correctly 1`] = `ReactWrapper {}`;
